pallets/foreign-investments/src/impls/mod.rs

```diff
@@ -187,19 +190,19 @@ impl<T: Config> ForeignInvestment<T::AccountId> for Pallet<T> {
 		who: &T::AccountId,
 		investment_id: T::InvestmentId,
 		foreign_payout_currency: T::CurrencyId,
-		pool_currency: T::CurrencyId,
-	) -> Result<ExecutedForeignCollectInvest<T::Balance>, DispatchError> {
+	) -> DispatchResult {
+		// Simply overwrite any existing payment currency to enable the investor to have
+		// full control over the payment currency
+		InvestmentPaymentCurrency::<T>::insert(who, investment_id, foreign_payout_currency);
```

Comment:
Isn't that criticial if something is happening currently? E.g.  swapping due to an `DecreaseInvestOrder`?

Reply:
Or a `IncreaseInvestOrder` with `payment_currency_a` followed by an `CollectInvest` with `payment_currency_b`

Reply:
[2768b4b](https://github.com/centrifuge/centrifuge-chain/pull/1556/commits/2768b4bdd81be4f3d025881252b75f6f0b538d08)